src/cynq/execution-graph/stream.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2024
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ */
+
+#include <chrono>              // NOLINT
+#include <condition_variable>  // NOLINT
+#include <cynq/execution-graph/stream.hpp>
+#include <mutex>  // NOLINT
+#include <queue>
+#include <thread>  // NOLINT
+
+namespace cynq {
+
+/**
+ * @brief Define the parameters for the ExecutionStream class
+ */
+struct ExecutionStreamParameters : public ExecutionGraphParameters {
```

Comment:
structs are public by default why would a child that inherits from a struct require the public keyword? this seems redundant.

Reply:
Yes, but this enforces that children structs can access to parent's publics.

https://www.programiz.com/cpp-programming/public-protected-private-inheritance